shared/crypto/keypair.go

```diff
@@ -33,14 +33,42 @@ type KeyPair interface {
 	// Marshalling
 	Marshal() ([]byte, error)
 	Unmarshal([]byte) error
+	MarshalJSON() ([]byte, error)
+	UnmarshalJSON([]byte) error
 }
 
 var _ KeyPair = &encKeyPair{}
 
 // encKeyPair struct stores the public key and the passphrase encrypted private key
 type encKeyPair struct {
-	PublicKey     PublicKey
-	PrivKeyArmour string
+	PublicKey     PublicKey `json:"public_key"`
+	PrivKeyArmour string    `json:"priv_key_armour"`
+}
+
+type simpleEncKeyPair struct {
+	PublicKey     string `json:"public_key"`
+	PrivKeyArmour string `json:"priv_key_armour"`
+}
+
+func (ekp *encKeyPair) MarshalJSON() ([]byte, error) {
```

Comment:
Bumping - just a NIT but these should be moved down next to the other marshalling methods